src/core/transport/chttp2_transport.c

```diff
@@ -763,6 +763,18 @@ static int contains_non_ok_status(
 
 static void do_nothing(grpc_exec_ctx *exec_ctx, void *arg, int success) {}
 
+static void fail_stream_op_locked(
+    grpc_exec_ctx *exec_ctx, grpc_chttp2_transport_global *transport_global,
+    grpc_chttp2_stream_global *stream_global, grpc_transport_stream_op *op) {
+  if (op->on_complete != NULL) {
+    grpc_exec_ctx_enqueue(exec_ctx, op->on_complete, 0);
+  }
+
+  if (op->on_cancel != NULL) {
+    grpc_exec_ctx_enqueue(exec_ctx, op->on_cancel, 0);
+  }
+}
```

Comment:
It feels like I should also queue the "op->recv_message_ready" closure with a failure (assuming it is set on the op) (because if a caller set the closure, the caller should be notified either of success or failure). I will explore this option more.